src/gt4py/next/program_processors/runners/dace_fieldview/transformations/gpu_utils.py

```diff
@@ -82,171 +86,253 @@ def gt_gpu_transformation(
+    block_size: Optional[Sequence[int | str] | str],
+    launch_bounds: Optional[int | str] = None,
+    launch_factor: Optional[int] = None,
+    **kwargs: Any,
 ) -> Any:
     """Set the block size related properties of _all_ Maps.
 
-    See `GPUSetBlockSize` for more information.
+    It supports the same arguments as `GPUSetBlockSize`, however it also has
+    versions without `_Xd`, these are used as default for the other maps.
+    If a version with `_Xd` is specified then it takes precedence.
 
     Args:
         sdfg: The SDFG to process.
-        gpu_block_size: The size of a thread block on the GPU.
+        block_size: The size of a thread block on the GPU.
         launch_bounds: The value for the launch bound that should be used.
         launch_factor: If no `launch_bounds` was given use the number of threads
             in a block multiplied by this number.
     """
-    xform = GPUSetBlockSize(
-        block_size=gpu_block_size,
-        launch_bounds=gpu_launch_bounds,
-        launch_factor=gpu_launch_factor,
-    )
-    return sdfg.apply_transformations_once_everywhere([xform])
-
-
-def _gpu_block_parser(
-    self: GPUSetBlockSize,
-    val: Any,
-) -> None:
-    """Used by the setter of `GPUSetBlockSize.block_size`."""
-    org_val = val
-    if isinstance(val, (tuple | list)):
-        pass
-    elif isinstance(val, str):
-        val = tuple(x.strip() for x in val.split(","))
-    elif isinstance(val, int):
-        val = (val,)
+    for dim in [1, 2, 3]:
+        for arg, val in {
+            "block_size": block_size,
+            "launch_bounds": launch_bounds,
+            "launch_factor": launch_factor,
+        }.items():
+            if f"{arg}_{dim}d" not in kwargs:
+                kwargs[f"{arg}_{dim}d"] = val
+    return sdfg.apply_transformations_once_everywhere(GPUSetBlockSize(**kwargs))
+
+
+def _make_gpu_block_parser_for(
+    dim: int,
+) -> Callable[["GPUSetBlockSize", Any], None]:
+    """Generates a parser for GPU blocks for dimension `dim`.
+
+    The returned function can be used as parser for the `GPUSetBlockSize.block_size_*d`
+    properties.
+    """
+
+    def _gpu_block_parser(
+        self: GPUSetBlockSize,
+        val: Any,
+    ) -> None:
+        """Used by the setter of `GPUSetBlockSize.block_size`."""
+        org_val = val
+        if isinstance(val, (tuple | list)):
+            pass
+        elif isinstance(val, str):
+            val = tuple(x.strip() for x in val.split(","))
+        elif isinstance(val, int):
+            val = (val,)
+        else:
+            raise TypeError(
+                f"Does not know how to transform '{type(org_val).__name__}' into a proper GPU block size."
+            )
+        if len(val) < dim:
+            raise ValueError(
+                f"The passed block size only covers {len(val)} dimensions, but dimension was {dim}."
+            )
+        if 0 < len(val) <= 3:
+            val = [*val, *([1] * (3 - len(val)))]
```

Comment:
It works below is a conversion to `int`.

Reply:
The case "tuple of string" is perfectly handled.
Assume we have `val = ("32", 4)` at line 181.
Because `len(val) <= 3` we will enter the `if` and we have something as
```
val = [*("32", 4), *([1] * (3 - len(val)))]
val = [ "32", 4, *([1] * 1)]
val = [ "32", 4, *([1])]
val = [ "32", 4, 1]
```
So we end up with different types, however, since the type conversion happens afterwards we end up with a list of integer.
What we could do, is moving the type conversion, up but it would not change anything.

Reply:
I see, the power of dynamic typing. The list elements are not used before the type conversion happens, hence no problem.